toolchain/check/function.h

```diff
@@ -27,13 +27,13 @@ struct SuspendedFunction {
 };
 
 // Checks that `new_function` has the same parameter types and return type as
-// `prev_function`, applying the specified set of substitutions to the
-// previous function. Prints a suitable diagnostic and returns false if not.
-auto CheckFunctionTypeMatches(Context& context,
-                              const SemIR::Function& new_function,
-                              const SemIR::Function& prev_function,
-                              Substitutions substitutions, bool check_syntax)
-    -> bool;
+// `prev_function`, or possibly a specific version of `prev_function`. Prints a
```

Comment:
..., or if `prev_specific_id` is specified, then the specific version of `prev_function`.